src/modules/sync/utils/htmlRenderers.ts

```diff
@@ -0,0 +1,638 @@
+  value: unknown,
+): ReplacePatchOperationValue | null => {
+  switch (typeof value) {
+    case "string":
+    case "boolean":
+    case "number":
+      return value;
+    case "object":
+      if (isCountLimitation(value)) {
+        return `${value.condition}: ${value.value}`;
+      }
+      if (isObjectReference(value)) {
+        return value.codename;
+      }
+      if (isObjectReferenceArray(value)) {
+        return value.map((r) => r.codename).join(", ");
+      }
+      if (isExternalIdReference(value)) {
+        return value.external_id;
+      }
+      if (isExternalIdReferenceArray(value)) {
+        return value.map((r) => r.external_id).join(", ");
+      }
+      if (isDependsOn(value)) {
+        return `${value.element.codename} ${value.snippet?.codename ? "of snippet " + value.snippet.codename : ""}`;
+      }
+      if (isDefaultValue(value)) {
+        return renderReplaceOpValue(value.global.value);
+      }
+      if (isMaximumTextLength(value)) {
+        return `${value.value} ${value.applies_to}`;
+      }
+      if (isValidationRegex(value)) {
+        return `<p><strong>Regex:</strong> ${value.regex}<strong></p><p>Flags:</strong> ${
+          value.flags ?? "—"
+        }</p><p><strong>IsActive:</strong> ${value.is_active || "—"}</p><p><strong>Validation message:</strong> ${
+          value.validation_message ?? "—"
+        }</p>`;
+      }
+      return getValueOrIdentifier(value);
+    default:
+      return null;
+  }
+};
+
+const renderDeletedEntity = (entityCodename: string) =>
+  `<div class="entity-detail"><div class="entity-name removed">${entityCodename}</div></div>`;
+
+const renderAddedTypesOrSnippetsSectionData = <T extends RequiredCodename<TypeOrSnippet>>(
+  diff: Pick<DiffObject<RequiredCodename<T>>, "added">,
+) => diff.added.map(a => renderAddedTypeOrSnippetData(a, a.elements)).join("\n");
+
+const renderAddedTaxonomiesSectionData = <T extends TaxonomyModels.IAddTaxonomyRequestModel>(
+  diff: Pick<DiffObject<RequiredCodename<T>>, "added">,
+) => diff.added.map(a => renderAddedTaxonomyData(a, a.terms)).join("\n");
+
+const renderDeletedEntitiesSectionData = <T extends { codename: string }>(
+  diff: Pick<DiffObject<RequiredCodename<T>>, "deleted">,
+) =>
+  `${
+    [...diff.deleted]
+      .map(renderDeletedEntity)
+      .join("\n")
+  }`;
+
+const renderUpdatedEntitiesSectionData = <T extends { codename: string }>(
+  diff: Pick<DiffObject<RequiredCodename<T>>, "updated">,
+) =>
+  [...diff.updated]
+    .filter(([, p]) => p.length > 0)
+    .flatMap(renderUpdatedEntity)
+    .join("\n");
+
+const renderUpdatedEntity = (
+  [entityCodename, patchOps]: Readonly<[string, Readonly<PatchOperation[]>]>,
+) =>
+  `<div class="entity-detail"><div class="entity-name" onClick="toggleVisibility('${entityCodename}')">${entityCodename}</div><div class="entity-operations" style="display: none" id="${entityCodename}">${
+    patchOps
+      .flatMap(renderPatchOp)
+      .join("\n")
```

Comment:
nitpick: this can be one lined to save some lines, but this one is up to preference :)

Reply:
oneline it is